stonesoup/kernel.py

```diff
@@ -16,17 +17,17 @@ class Kernel(Base):
     @abstractmethod
     def __call__(self, state1, state2=None):
```

Comment:
now that type of state1 and state2 is now StateVectors, would it make sense to change the name of these variables as to note be confused with the State type?

Reply:
Thinking about this more, as you highlight it @hpritchett-dstl ...

I do wonder if it was better being states passed, as before. This mirrors the `Measure` class, and also `Model.function()` methods. Could there be other non state vector based information as part of the state that could be used in a kernel (metadata, Gaussian covariance, ...)?